File: server/src/entity/Bug.ts
import { BaseEntity, Column, CreateDateColumn, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn, UpdateDateColumn } from "typeorm";
import { User } from "./User";

@Entity("bug")
export class Bug extends BaseEntity {


    @PrimaryGeneratedColumn()
    id: string

    @Column()
    name: string

    @Column()
    details: string

    @Column()
    version: string

    @Column()
    priority: number

    @ManyToOne(() => User, user => user.bugs)
    @JoinColumn({name: "creatorId"})
    creator: User

    @Column({nullable: true})
    creatorId?: string
    
    @Column({default: false})
    is_completed: boolean

    @Column({nullable: true, default: null})
    resolution?: string

    @CreateDateColumn()
    created_at: Date

    @UpdateDateColumn()
    updated_at: Date
}

File: client/app/src/main/java/com/ben/bugtrackerclient/model/Bug.kt
package com.ben.bugtrackerclient.model


import android.os.Parcelable
import androidx.versionedparcelable.ParcelField
import androidx.versionedparcelable.VersionedParcelize
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.util.*

@Parcelize
data class Bug(
    @SerializedName("created_at")
    val createdAt: String?,
    val details: String?,
    val id: Int?,
    @SerializedName("is_completed")
    val isCompleted: Boolean,
    val name: String?,
    val resolution: String?,
    val priority: Int?,
    @SerializedName("updated_at")
    val updatedAt: String?,
    val version: String?,
    val creator: User? = null
) : Parcelable